hopsworksai/provider.go

```diff
@@ -39,6 +42,32 @@ func Provider(version string) func() *schema.Provider {
 					Sensitive:   true,
 					DefaultFunc: schema.EnvDefaultFunc("HOPSWORKSAI_API_KEY", ""),
 				},
+				"api_gateway": {
+					Description: "URL of the API Gateway to use.",
+					Type:        schema.TypeString,
+					Optional:    true,
+					Default:     api.DEFAULT_API_GATEWAY,
+					ValidateDiagFunc: func(v interface{}, path cty.Path) diag.Diagnostics {
+						value := v.(string)
+						u, err := url.Parse(value)
+						var diagnostics diag.Diagnostics
+						if err != nil {
+							diagnostic := diag.Diagnostic{
+								Severity: diag.Error,
+								Summary:  "Could not parse API Gateway URL",
+								Detail:   fmt.Sprintf("Failed to parse URL: %s", err),
+							}
+							diagnostics = append(diagnostics, diagnostic)
+						} else if u.Scheme == "" {
+							diagnostic := diag.Diagnostic{
+								Severity: diag.Error,
+								Summary:  "API Gateway URL is missing scheme http/https",
+							}
+							diagnostics = append(diagnostics, diagnostic)
+						}
+						return diagnostics
```

Comment:
Can you add a Warning diagnostics regardless that this is is only intended for development purposes.